Bottom: 36e86064b1de5d4dcfe5b7395e0ec7814a4066f7
Top:    34779ae77b64e2e3ac2f6981df4f7e52e915b871
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-22 02:10:15 +0000

XXX: THIS IS NOT TO BE EDITED, JUST A CHERRY PICKED COPY FOR DEBUG... Add polygon parentage tree debug view

NB: Also contains some probably bug fixes for PBO operations,
in particular - relating to bad exit codes when a NULL poly is
copied.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2372153..1157244 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2895,10 +2895,8 @@ poly_Boolean_free (POLYAREA * ai, POLYAREA * bi, POLYAREA ** res, int action)
   *res = NULL;
 
   /* Make copies for tracking polygon parentage (DEBUG) */
-//  if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
-//      return err_no_memory;
-  a_copy = NULL;
-  b_copy = NULL;
+  if (!poly_M_Copy0 (&a_copy, a) || !poly_M_Copy0 (&b_copy, b))
+      return err_no_memory;
 
   /* Move the parentage information over onto the copy */
   if (a_copy != NULL)
